controllers/client/openstackclient_controller.go

```diff
@@ -307,17 +308,23 @@ func (r *OpenStackClientReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 	spec := openstackclient.ClientPodSpec(ctx, instance, helper, configVarsHash)
 
+	podSpecHash, err := util.ObjectHash(spec)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	podSpecAnnotation := labels.GetGroupLabel("openstackclient") + "/pod-spec-hash"
+
 	op, err := controllerutil.CreateOrPatch(ctx, r.Client, osclient, func() error {
 		isPodUpdate := !osclient.ObjectMeta.CreationTimestamp.IsZero()
-		if !isPodUpdate {
+		currentPodSpecHash, hasPodSpecHash := osclient.Annotations[podSpecAnnotation]
+		if !isPodUpdate || (hasPodSpecHash && currentPodSpecHash != podSpecHash) {
```

Comment:
first time this reconciles it will not update the spec, but will add the hash annotation. From then on it will update the spec when the hash changes (which will almost definitely result in a forbidden error)

my only concern is that the spec could have changed for that initial reconcile, so maybe worth asserting the spec in that case too...